hls-test-utils/src/Test/Hls.hs

```diff
@@ -712,6 +717,17 @@ setHlsConfig config = do
   -- requests!
   skipManyTill anyMessage (void configurationRequest)
 
+captureKickDiagnostics :: Session [Diagnostic]
+captureKickDiagnostics = do
+    _ <- skipManyTill anyMessage nonTrivialKickStart2
+    messages <- manyTill anyMessage nonTrivialKickDone2
```

Comment:
Would it work if we are using the combination of `nonTrivialKickDone` and `nonTrivialKickDone2` work?

Reply:
Sorry, one of them is `nonTrivialKickStart`. But never mind, I think it won't work anyway.
To make sure it works we would have to customize a signal for every rule. But It should only turned on if we enable it(Provide a switch as we did for ideTesting🤔), otherwise there would be too many rules being logged.

Reply:
We don't even need to change the method name, we could put the information about what rule it was in the method parameters.

Or we could just use `$/logTrace`, which we don't use at all at the moment and is maybe quite suited to this?